### PR TITLE
ci: add shfmt

### PR DESCRIPTION
**Pull request by ggilmore**
This PR adds [shfmt](https://github.com/mvdan/sh#shfmt), a shell script formatter, as a CI step that will run on all pull requests. If your change includes a script that doesn't conform to our style guide, a comment will be posted on your PR that highlights the changes that need to be made.

I setup shfmt to use flags that conform to Google's style guide. This is the same style that we use in github.com/sourcegraph/sourcegraph.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -25,7 +25,18 @@ jobs:
...
+  shfmt:
+    name: shfmt
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: reviewdog/action-shfmt@v1.0.2
+        with:
+          filter_mode: "nofilter"
+          fail_on_error: "true"
+          shfmt_flags: "-i 2 -ci -bn"
```
Comment (ggilmore) [inline .github/workflows/ci.yml:38]: Flags from https://google.github.io/styleguide/shellguide.html#s5-formatting

Comment (stefanhengl) [reply .github/workflows/ci.yml:38]: Couldn't find the flags in the doc you reference. I found [this](https://github.com/mvdan/sh/blob/3c0f553b3180d01710580241fcbf5f89a87f48d3/cmd/shfmt/shfmt.1.scd#examples) which seems to match the flags you chose.

I guess you are planning to submit some scripts :-)

**Review by stefanhengl — APPROVED** [review]
nit: Could you add a tiny PR description?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (keegancsmith) [reply .github/workflows/ci.yml:38]: yeah a script so we can locally run shfmt with the same setup would be great :) Is there maybe a config file so we can just run shfmt?

**Pull request merged**